core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/ZippedStream.scala

```diff
@@ -25,24 +44,26 @@ class ZippedStream[A, B](private val parentA: Stream[A], private val parentB: St
     case _ => None
   }
 
-  private[this] var aSlot: Option[A] = None
-  private[this] var bSlot: Option[B] = None
+  private[this] var aSlot: Option[(A, Time)] = None
+  private[this] var bSlot: Option[(B, Time)] = None
 
   def attemptPublish(): Unit = {
-    if (aSlot.isDefined && bSlot.isDefined) {
-      publishValue((aSlot.get, bSlot.get))
+    if (aSlot.isDefined && bSlot.isDefined &&
+      // if the streams are synced, make sure the timestamps match
+      (aSlot.get._2 == bSlot.get._2 || expectedPeriodicity == NonPeriodic)) {
```

Comment:
This is going to cause issues when we use syncTo. For example, the time stamp when we get the camera data will never be the same as the gyro data ever

Reply:
Also, please move the check into a separate line. Maybe define a `val` or method

Reply:
In the case with the camera data, the expectedPeriodicity will be NonPeriodic, so the check is skipped!